tools/squidclient/squidclient.cc

```diff
@@ -555,13 +544,13 @@ main(int argc, char *argv[])
 
         /* Send the HTTP request */
         debugVerbose(2, "Sending HTTP request ... ");
-        bytesWritten = Transport::Write(msg, strlen(msg));
+        bytesWritten = Transport::Write(messageHeader.c_str(), messageHeader.length());
 
         if (bytesWritten < 0) {
             std::cerr << "ERROR: write" << std::endl;
             exit(EXIT_FAILURE);
-        } else if ((unsigned) bytesWritten != strlen(msg)) {
-            std::cerr << "ERROR: Cannot send request?: " << std::endl << msg << std::endl;
+        } else if ((unsigned) bytesWritten != messageHeader.length()) {
```

Comment:
Please use C++ static_cast and either cast to size_t (to better match std::string::length() result) or at least spell out "unsigned int" fully.

Reply:
done